source/base/partitioner.cc

```diff
@@ -194,12 +194,14 @@ namespace Utilities
         }
 
       types::global_dof_index my_size = local_size();
-      // Allow non-zero start index for the vector. send this data to all
-      // processors
+
+      // Allow non-zero start index for the vector. Part 1:
+      // Assume for now that the index set of rank 0 starts with 0
+      // and therefore has an increased size.
       if (my_pid == 0)
         my_size += local_range_data.first;
 
-      types::global_dof_index my_shift = 0;
+      types::global_dof_index my_shift;
```

Comment:
I prefer that you keep the initialization to zero here to be on the safe side regarding reading undefined data when moving code around.